### ISSUE TITLE
Enable Broader IBC Optimization for Libraries

### ISSUE THREAD

**Issue by brianrob** [dotnet/corefx#30679]
Today there are a handful of libraries that get IBC optimized.  We should look at broadening this set.

**Comment by brianrob** [comment]
@HFadeel, IBC is a profile guided optimization technology for managed code.  You can read a bit more about it here: https://docs.microsoft.com/en-us/dotnet/framework/tools/mpgo-exe-managed-profile-guided-optimization-tool

Note that this doc refers to the technology as MPGO, which is equivalent to IBC, it's just the public name.

**Comment by HFadeel** [comment]
@brianrob thanks for your answer. I didn't know IBC is another name for MPGO.

**Comment by joperezr** [comment]
@brianrob is this needed for 3.0?

**Comment by brianrob** [comment]
@joperezr, we should keep this for 3.0.  There is some work ongoing in this area.

**Comment by joperezr** [comment]
Sounds good, I'm currently triaging the area and just wanted to make sure that all 3.0 work is being either worked on or scheduled. Should I assign this one to you or is there somebody else doing the work?

**Comment by brianrob** [comment]
@MichalStrehovsky is doing the work here so you can assign it to him.  @MichalStrehovsky if you have a CoreFX issue for your work we can dupe it.

**Comment by MichalStrehovsky** [comment]
What work is this actually tracking? Broadening the set of assemblies we collect IBC data for is not something I've been looking into.

Cc @sergiy-k

**Comment by sergiy-k** [comment]
I agree with @MichalStrehovsky. It's not clear to me what exactly this is about and whether it aligns with the work that @MichalStrehovsky is doing.
It seems that this work implies adding new training scenarios that should cover additional libraries.

**Comment by joperezr** [comment]
@brianrob ping. Do you mind answering @sergiy-k question above? If this is not tracking 3.0 work I would like to either move it to Future or close it.
